lib/App/DuckPAN/Cmd/Test.pm

```diff
@@ -70,14 +70,38 @@ sub run {
 				if (-f $path) {
 					$ENV{'DDG_TEST_FATHEAD'} = $id;
 					push @to_test, "t/validate_fathead.t";
-				} else {
+				}
+				else {
 					$self->app->emit_and_exit(1, "Could not find output.txt for $id in $path");
 				}
 			}
 
 			$self->app->emit_and_exit(1, "Could not find any tests for $id $ia_type") unless @to_test;
 		};
 
+		# For the front-end tests
+		print("\n\nSearching for front-end tests\n");
+		if(scalar(@args) == 0) {
+			$self->app->emit_error("Couldn't run front-end tests. Please ensure that jasmine is installed globally.\n\nnpm install -g jasmine-node\n") if system("jasmine-node --test-dir spec/");
+		}
+		else {
+			foreach my $ia_name (@args) {
+
+				if($ia_type eq "Goodie" or $ia_type eq "Spice") {
```

Comment:
IA type is set based on the repo duckpan is running inside. We can do this check once. Replace the `else` with and `elsif` for this condition -- I think that'll work...